### DISCUSSION TITLE
NextRequest geo region no longer available

### DISCUSSION THREAD

**Discussion by jpallares** [vercel/next.js#54518] [Help]
### Summary

I usualy got the NextRequest geo.region with the typical middleware:

```ts
export async function middleware(nextRequest: NextRequest) {
  const { nextUrl: url, geo } = nextRequest;

  let region = geo.region || 'XX';

  url.searchParams.set(regionCodeParam, region);

  return NextResponse.rewrite(url);
}
```

But at some moment it stopeed working. I was not aware, a user told me.

I've been debugging and the geo.country arrives (ES for Spain in may case) but no region nor city.

Tried with headers ('x-vercel-ip-country', 'x-vercel-ip-country-region', etc) but no luck.

I'm testing deploying to a branch and logging the retrieved values.

What could be the reason?

No recent changes in the code in that area.

**Comment by ndueber** [comment]
I also have this issue and looking to see how i can get latitude and longitude and put it in the response headers.

**Reply by ndueber** [reply]
https://github.com/vercel/next.js/discussions/39619

Maybe this has the answer. maybe its bc  we are testing in a local environment
